drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/editor/ScenarioSimulationEditorPresenter.java

```diff
@@ -470,6 +499,8 @@ public void getModelSuccessCallbackMethod(DataManagementStrategy dataManagementS
         view.setContent(model.getSimulation());
         context.getStatus().setSimulation(model.getSimulation());
         CustomBusyPopup.close();
+        // check if structure is valid
+        getValidateCommand().execute();
```

Comment:
@danielezonca 
I think this specific back-and-forth may and should be avoided. We can add those "FactMappingError" inside "ScenarioSimulationModel", mark them to be ignored by Xstream (=not persisted on xml) and showing them at load, without having to issue another service call. Nothing would prevent, anyway, to let the user call the "validate" service at will.

Reply:
I prefer to postpone this refactor so I have opened this ticket to track it
https://issues.jboss.org/browse/DROOLS-4435